### PR TITLE
[awsfargate] Add undocumented fields 

### LINKED ISSUE #31129: [awsfargate] Fields are not documented and no TestMetricsetFieldsDocumented
Thanks to @zmoog finding this issue: 

Fields are not documented in `awsfargate` metricbeat module and also the `TestFetch` integration test to catch undocumented fields is not added to this module either. 

Example of `TestFetch`: https://github.com/elastic/beats/blob/main/x-pack/metricbeat/module/aws/cloudwatch/cloudwatch_integration_test.go#L30

### PR DESCRIPTION
**Pull request by zmoog**
## What does this PR do?

This PR adds a few missing mapping in the `fields.yml` files at the package and data stream level. 

To avoid this issue in the future, we also added a test to make sure every element in the `data.json` file has a proper mapping in the `fields.yml` file.

## Why is it important?

All fields must be documented  and properly mapped for the best possible indexing in Elasticsearch.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

- Fixes #31129

### CODE REVIEW

File: x-pack/metricbeat/module/awsfargate/task_stats/task_stats_integration_test.go [hunk 1]
```diff
@@ -65,7 +66,14 @@ func TestData(t *testing.T) {
...
-	standardizeEvent := m.StandardizeEvent(event)
 
+	// Build a metricset to test the event
+	metricSet := mbtest.NewReportingMetricSetV2Error(t, config)
+
+	// The goal here is to make sure every element inside the
+	// event has a matching field ("no field left behind").
+	mbtest.TestMetricsetFieldsDocumented(t, metricSet, []mb.Event{event})
+
+	standardizeEvent := m.StandardizeEvent(event)
```
Comment (zmoog) [inline x-pack/metricbeat/module/awsfargate/task_stats/task_stats_integration_test.go:77]: @kaiyan-sheng I started adding the check for the undocumented fields here (we already loaded the event from the JSON files). 

It works, but it's probably worth adding a proper `TestFetch()` for this kind of check, as you originally suggested, isn't it? What do you think?

Comment (kaiyan-sheng) [reply x-pack/metricbeat/module/awsfargate/task_stats/task_stats_integration_test.go:78]: Not sure if it has to be this way, but usually we have TestFetch function for TestMetricsetFieldsDocumented and TestData to WriteEvents.

**Review by kaiyan-sheng — APPROVED** [review]

File: x-pack/metricbeat/module/awsfargate/task_stats/task_stats_integration_test.go (hunk 1 above)
Comment (zmoog) [reply x-pack/metricbeat/module/awsfargate/task_stats/task_stats_integration_test.go:78]: You're right: I am moving the `TestMetricsetFieldsDocumented()` to the `TestFetch` function; this case isn't special enough to make an exception.

Comment (zmoog) [reply x-pack/metricbeat/module/awsfargate/task_stats/task_stats_integration_test.go]: Moved!

**Pull request merged**